packages/eslint-plugin-svelte/src/utils/svelte-context.ts

```diff
@@ -0,0 +1,259 @@
+import type { RuleContext } from '../types.js';
+import fs from 'fs';
+import path from 'path';
+import { getPackageJsons } from './get-package-json.js';
+import { getFilename, getSourceCode } from './compat.js';
+
+const isRunInBrowser = !fs.readFileSync;
+
+export type SvelteContext = (
+	| ({
+			svelteVersion: '3/4';
+	  } & {
```

Comment:
Why is this split into 2?

Reply:
For more strict typing. Reinforcing the types eliminates improbable patterns.

Reply:
I don't understant, I thought `{a: string} & {b: int}` is equal to {a: string; b: int}`, isn't it?

Reply:
updated: https://github.com/sveltejs/eslint-plugin-svelte/pull/980/commits/0e28cc19edc9b33c6d96abe64bced627f7851e1f